.dotnet/src/Custom/Files/FileClient.Protocol.cs

```diff
@@ -209,4 +209,74 @@ public virtual ClientResult DownloadFile(string fileId, RequestOptions options)
         using PipelineMessage message = CreateDownloadFileRequest(fileId, options);
         return ClientResult.FromResponse(_pipeline.ProcessMessage(message, options));
     }
+
+    /// <summary>
+    /// <para>[Protocol Method]</para>
+    /// Creates a new <see cref="IncrementalUploadJob"/> that can be used to incrementally create a file, uploading
+    /// individual data parts with arbitrary ordering and parallelism before being assembled by a final completion.
+    /// </summary>
+    /// <param name="content">
+    /// <para>
+    /// The request content describing the job to create. This is a JSON object with required <c>filename</c>,
+    /// <c>purpose</c>, <c>bytes</c>, and <c>mime_type</c> properties.
+    /// </para>
+    /// Example:
+    /// <code>
+    /// BinaryContent.Create(BinaryData.FromString($$"""
+    ///     {
+    ///         "purpose": "fine-tune",
+    ///         "filename": ""{{filenameToCreate}}"",
+    ///         "bytes": ""{{totalSizeOfFile}}"",
+    ///         "mime_type": "text/jsonl"
+    ///     }));
+    /// </code>
+    /// </param>
+    /// <param name="options">
+    /// Additional options for the protocol request, including cancellation and error behavior.
+    /// </param>
+    /// <returns>
+    /// A new instance of <see cref="IncrementalUploadJob"/> that can be used to add data parts and complete the upload.
+    /// A successfully completed upload will provide an assembled file with an ID that can be used in other operations.
+    /// </returns>
+    public virtual async Task<IncrementalUploadJob> CreateIncrementalUploadJobAsync(BinaryContent content, RequestOptions options)
```

Comment:
This looks like what we discussed with @KrzysztofCwalina -- thanks for making it happen! 👍🏻

Reply:
Of course! And thank you for enduring all my whinging as we figured out to balance the right approach.